devcontainers/build.py

```diff
@@ -0,0 +1,223 @@
+            print(f'creating symlink for path \'{path}\'')
+        if path is None:
+            sys.exit('no supported CMake configs detected')
+        symlink = pathlib.Path.cwd().joinpath('compile_commands.json')
+        if symlink.is_symlink():
+            symlink.unlink()
+        pathlib.Path.cwd().joinpath('compile_commands.json').symlink_to(path)
+
+    def _run(self: 'Routines', command: typing.List[str]) -> None:
+        print('running command: ' + ' '.join(command))
+        code = subprocess.run(command, encoding='UTF-8', stderr=subprocess.STDOUT, env=os.environ).returncode
+        if code:
+            sys.exit(f'error: subprocess failed: {errno.errorcode[code]} (code: {code})')
+
+    def _decorate_cmake_variable(self: 'Routines', var: str, value: str, type: typing.Union[str, None] = None) -> str:
+        if type is not None:
+            return f'-D{var.upper()}:{type}={value}'
+        return f'-D{var.upper()}={value}'
+    
+    def _decorate_conan_profile_entry(self: 'Routines', section: str, entry: str, value: str) -> str:
+        return f'--{section}={entry}={value}'
+
+
+def parse_cli_args() -> argparse.Namespace:
+    parser = argparse.ArgumentParser(description=description, formatter_class=argparse.RawDescriptionHelpFormatter)
+    # Routines
+    parser.add_argument('-b', '--build', action='store_true', dest='build',
+        help='runs cmake build, runs after configure.'
+    )
+    parser.add_argument('-c', '--configure', action='store_true', dest='configure',
+        help='runs cmake configure. If CMakeUserPresets.json is preset, uses conan presets. '
+        'Runs after install.'
+    )
+    parser.add_argument('-i', '--install', action='store_true', dest='install',
+        help='runs conan install with devcontainers/conanfile.py. If you need specific profile, '
+        '(other than default) specify it with --profile, some examples are located under '
+        'devcontainers/templates as .ini files.'
+    )
+    parser.add_argument('-r', '--remove', action='store_true', dest='remove',
+        help='removes subdirectories in build/, according to configs provided. Might be useful '
+        'if CMake fresh configuration is required. Always executed before any other routine.'
+    )
+    parser.add_argument('-l', '--link', action='store_true', dest='link',
+        help='creates symlink to build/.../compile_commands.json in source directory. In case '
+        'both configs specified, links to Debug. In case symlink already exists, rewrites it.'
+    )
+    # Environment
+    parser.add_argument('--build-dir', action='store', dest='build_directory', 
+        help='specify root directory to put build files in. Defaults to \'build\'.'
+    )
+    parser.add_argument('--config', action='append', dest='configs', choices=['Debug', 'Release'],
+        help='specify configs for build, options are Debug and Release.'
+    )
+    parser.add_argument('--parallel', action='store', dest='cores', 
```

Comment:
You might want to add an option for CMake generators, e.g. Ninja handles parallel builds quite nicely.

Edit: Or just set the "CMAKE_GENERATOR" environment variable and keep this script as simple as possible!